pkg/yurthub/server/nonresource_test.go

```diff
@@ -0,0 +1,174 @@
+	remoteServers := map[string]int{"https://10.10.10.113:6443": 2}
+	u, _ := url.Parse("https://10.10.10.113:6443")
+	fakeHealthchecker := healthchecker.NewFakeChecker(true, remoteServers)
+	dStorage, err := disk.NewDiskStorage(testDir)
+	defer func() {
+		if err := os.RemoveAll(testDir); err != nil {
+			t.Errorf("Unable to clean up test directory %q: %v", testDir, err)
+		}
+	}()
+
+	// store the kubelet ca file
+	caFile := filepath.Join(testDir, "ca.crt")
+	if err := dStorage.Create("ca.crt", certificatePEM); err != nil {
+		t.Fatalf("Unable to create the file %q: %v", caFile, err)
+	}
+
+	// store the kubelet-pair.pem file
+	pairFile := filepath.Join(testDir, "kubelet-pair.pem")
+	pd := bytes.Join([][]byte{certificatePEM, keyPEM}, []byte("\n"))
+	if err := dStorage.Create("kubelet-pair.pem", pd); err != nil {
+		t.Fatalf("Unable to create the file %q: %v", pairFile, err)
+	}
+
+	// store the yurthub-current.pem
+	yurthubCurrent := filepath.Join(testDir, "yurthub-current.pem")
+	if err := dStorage.Create("yurthub-current.pem", pd); err != nil {
+		t.Fatalf("Unable to create the file %q: %v", yurthubCurrent, err)
+	}
+
+	// set the YurtHubConfiguration
+	cfgNew := &config.YurtHubConfiguration{
+		RootDir:               testDir,
+		RemoteServers:         []*url.URL{u},
+		KubeletRootCAFilePath: caFile,
+		KubeletPairFilePath:   pairFile,
+	}
+
+	cfgNew.CertMgrMode = "hubself"
+	certMgr, err := hubself.NewFakeYurtHubCertManager(testDir, yurthubCon, string(certificatePEM), string(keyPEM))
+	if err != nil {
+		t.Fatalf("error in cfg %v", err)
+	}
+	certMgr.Start()
+	rcm, err := rest.NewRestConfigManager(cfgNew, certMgr, fakeHealthchecker)
+	if err != nil {
+		t.Errorf("failed to create RestConfigManager: %v", err)
+	}
+
+	rc := rcm.GetRestConfig(true)
+	clientSet, err := kubernetes.NewForConfig(rc)
+
+	testcases := map[string]struct {
+		Accept      string
+		Verb        string
+		Path        string
+		Code        int
+		ContentType string
+	}{
+		"version info": {
+			Accept:      "application/json",
+			Verb:        "GET",
+			Path:        "/version",
+			Code:        http.StatusOK,
+			ContentType: "application/json",
+		},
+
+		"discovery v1": {
+			Accept:      "application/json",
+			Verb:        "GET",
+			Path:        "/apis/discovery.k8s.io/v1",
+			Code:        http.StatusOK,
+			ContentType: "application/json",
+		},
+		"discovery v1beta1": {
+			Accept:      "application/json",
+			Path:        "/apis/discovery.k8s.io/v1beta1",
+			Code:        http.StatusOK,
+			ContentType: "application/json",
+		},
+	}
+
+	for k, tt := range testcases {
+		t.Run(k, func(t *testing.T) {
+			req, _ := http.NewRequest(tt.Verb, tt.Path, nil)
+			if len(tt.Accept) != 0 {
+				req.Header.Set("Accept", tt.Accept)
+			}
+
+			req.RemoteAddr = "127.0.0.1"
+
+			var handler http.Handler = http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+
+			})
+			wrapNonResourceHandler(handler, cfgNew, clientSet)
+
```

Comment:
why the process results(like cache, status code etc.) of `wrapNonResourceHandler` are not verified?

Reply:
The error will be returned if wrapNonResourceHandler encountered errors. And I will add cache verify to ensure the unit test.

Reply:
but the `tt.code` field is unused.

Reply:
OK, and I will check the code.